mlir/lib/Conversion/NVGPUToNVVM/NVGPUToNVVM.cpp

```diff
@@ -53,6 +55,16 @@ static Value truncToI32(ConversionPatternRewriter &rewriter, Location loc,
   return rewriter.create<LLVM::TruncOp>(loc, rewriter.getI32Type(), value);
 }
 
+/// Returns warp-size as a value.
+static Value getWarpSizeValue(ImplicitLocOpBuilder &b) {
```

Comment:
Is there an interest in having a singleton here as opposed to just storing the result in a variable in the related place?

Reply:
Second thing, the implementation of the singleton is going to break whenever we run the code on more than one function, since the builder will change and we'll still return the value of the very first one. (Which at this point could have been destroyed, reassigned, etc.)

Long story short, drop the singleton construct :).

Reply:
it was a bad idea to put a singleton. I thought with my previous compiler's logic, my bad. removed